Core/clim-core/presentation-translators.lisp

```diff
@@ -483,14 +483,23 @@ and used to ensure that presentation-translators-caches are up to date.")
          t)))
 
 (defun map-applicable-translators (func presentation input-context frame window x y
-                                   &key event (modifier-state 0) for-menu button)
+                                   &key event (modifier-state 0) for-menu button
+                                   &aux (processed nil))
   (labels ((process-presentation (context presentation)
              (let* ((context-ptype (first context))
-                    (maybe-translators
-                     (find-presentation-translators (presentation-type presentation)
-                                                    context-ptype
-                                                    (frame-command-table frame))))
-               (loop for translator in maybe-translators
+                    (maybe-translators (find-presentation-translators
+                                        (presentation-type presentation)
+                                        context-ptype
+                                        (frame-command-table frame))))
+               (setf maybe-translators
+                     (remove-if (lambda (translator)
+                                  (member (cons translator presentation)
+                                          processed
+                                          :test #'equal))
+                                maybe-translators))
+               (loop
+                  for translator in maybe-translators
```

Comment:
would 
```lisp
for key = (cons translator presentation)
unless (member key processes :test #'equal)
do (push key processed)
and …
```
instead of the `remove-if` work? Seems simpler and less consy.
```

Reply:
I'm a poor loop reader, that would be
```
unless
do
and
when
do
```

how about

```
(let* ((context-ptype (first context))
                    (maybe-translators (find-presentation-translators
                                        (presentation-type presentation)
                                        context-ptype
                                        (frame-command-table frame))))
               (dolist (translator maybe-translators)
                 (let ((key (cons translator presentation)))
                   (unless (member key processed :test #'equal)
                     (push key processed)
                     (when (and (or (not for-menu) (eql for-menu (menu translator)))
                                (test-presentation-translator
                                 translator presentation context-ptype
                                 frame window x y
                                 :event event :modifier-state modifier-state
                                 :for-menu for-menu :button button))
                       (funcall func translator presentation context))))))
```

Reply:
Fine with me.

Reply:
fixed